tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/GZipUtilities.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http.Headers;
+using Microsoft.AspNetCore.Http;
+
+namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    /// <summary>
+    /// Utility methods to compress and decompress content to/from GZip.
+    /// </summary>
+    public static class GZipUtilities
+    {
+        private const string Gzip = "gzip";
+        private const string ContentEncoding = "Content-Encoding";
+
+        public static byte[] CompressBody(byte[] incomingBody, IDictionary<string, string[]> headers)
+        {
+            if (headers.TryGetValue(ContentEncoding, out var values) && values.Contains(Gzip))
+            {
+                return CompressBodyCore(incomingBody);
+            }
+
+            return incomingBody;
+        }
+
+        public static byte[] CompressBody(byte[] incomingBody, IHeaderDictionary headers)
+        {
+            if (headers.TryGetValue(ContentEncoding, out var values) && values.Contains(Gzip))
+            {
+                return CompressBodyCore(incomingBody);
+            }
+
+            return incomingBody;
+        }
+
+        public static byte[] CompressBodyCore(byte[] body)
+        {
+            using (var uncompressedStream = new MemoryStream(body))
+            using (var resultStream = new MemoryStream())
+            {
+                using (var compressedStream = new GZipStream(resultStream, CompressionMode.Compress))
+                {
+                    uncompressedStream.CopyTo(compressedStream);
```

Comment:
Don't we need to Flush the stream before it writes the the bytes as compressed?

Reply:
I don't believe so?

`CopyTo` seems to [advance both automatically](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.copyto?view=net-6.0).

Reply:
Advancing is not the issue - GZipStream only actually compresses the bytes in the stream after Flush is called.

Reply:
I thought this was only in .NET Framework?

Reply:
Doesn't look like Flush does much - https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/sys/system/IO/compression/DeflateStream.cs#L294

We may need to call Dispose though - https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/sys/system/IO/compression/DeflateStream.cs#L605

Reply:
Actually, that is the .NET Framework issue where you need to call Dispose. In .NET Core+, we just need to call Flush. I will fix this.